services/rfq/relayer/service/relayer.go

```diff
@@ -297,35 +299,62 @@ func (r *Relayer) startCCTPRelayer(ctx context.Context) (err error) {
 	return nil
 }
 
-func (r *Relayer) processDB(parentCtx context.Context) (err error) {
-	ctx, span := r.metrics.Tracer().Start(parentCtx, "processDB")
+func (r *Relayer) processDB(ctx context.Context) (err error) {
+	ctx, span := r.metrics.Tracer().Start(ctx, "processDB")
 	defer func() {
+		r.recordDBStats(ctx, span)
 		metrics.EndSpanWithErr(span, err)
 	}()
 
 	requests, err := r.db.GetQuoteResultsByStatus(ctx, reldb.Seen, reldb.CommittedPending, reldb.CommittedConfirmed, reldb.RelayCompleted, reldb.ProvePosted, reldb.NotEnoughInventory)
 	if err != nil {
 		return fmt.Errorf("could not get quote results: %w", err)
 	}
+
+	g, ctx := errgroup.WithContext(ctx)
 	// Obviously, these are only seen.
-	for _, request := range requests {
-		// if deadline < now
-		if request.Transaction.Deadline.Cmp(big.NewInt(time.Now().Unix())) < 0 && request.Status.Int() < reldb.RelayCompleted.Int() {
-			err = r.db.UpdateQuoteRequestStatus(ctx, request.TransactionID, reldb.DeadlineExceeded)
+	for _, req := range requests {
+		request := req // capture func literal
+		g.Go(func() error {
```

Comment:
Think it's a very bad idea not to have *any kind of* limit on number of go routines we spawn, seven if we keep it at 50.

This doesn't have to be a lot of code, just sem.Acquire and sem.release like we do in ethergo/chain/client